codalab/worker/worker.py

```diff
@@ -260,17 +260,22 @@ def check_num_runs_stop(self):
 
     def start(self):
         """Return whether we ran anything."""
-        self.load_state()
-        self.sync_state()
-        self.image_manager.start()
-        if not self.shared_file_system:
-            self.dependency_manager.start()
+        try:
+            self.load_state()
+            self.sync_state()
+            self.image_manager.start()
+            if not self.shared_file_system:
+                self.dependency_manager.start()
+        except Exception as ex:
+            # if we encounter any problem of starting up before it goes to the loop, we still want to cleanup.
+            logger.error(ex)
+            self.cleanup()
         while not self.terminate:
             try:
                 self.process_runs()
                 self.save_state()
                 self.checkin()
-                self.check_termination()
+                self.check_termination()  # if terminate_and_restage and no restaged bundles, then it will terminate.
```

Comment:
You should move this comment to be a docstring within the `check_termination` method definition itself.

Reply:
Oh this comment was actually a note for me to understand the logic. It's not meant to be there. Sorry